### ISSUE TITLE
ValuesAttribute uses an outdated version of TestCaseAttribute's type coercion logic

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit#2638]
As part of fixing https://github.com/nunit/nunit/issues/2630, I copied many tests from TestCaseAttributeTests to ValueAttributeTests and had to remove them again because they failed.

https://github.com/nunit/nunit/blob/c79e1c0a49c6c7fe1b2f895116ba02d93d0fe91e/src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs#L427-L432

When translated to ValuesAttribute:

```cs
[Test]
public void CanConvertIntToNullableLong([Values(1)] long? x)
{
    Assert.That(x.HasValue);
    Assert.That(x.Value, Is.EqualTo(1));
}
```

Gives:

> ```
> System.ArgumentException : Object of type 'System.Int32' cannot be converted to type 'System.Nullable`1[System.Int64]'.
>    at System.RuntimeType.CheckValue(Object value, Binder binder, CultureInfo culture, BindingFlags invokeAttr)
>    at System.Reflection.MethodBase.CheckArguments(Object[] parameters, Binder binder, BindingFlags invokeAttr, CultureInfo culture, Signature sig)
>    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture, Boolean skipVisibilityChecks)
>    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
>    at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
>    at NUnit.Framework.Internal.Reflect.InvokeMethod(MethodInfo method, Object fixture, Object[] args)
> ```

Before @ChrisMaddock brought https://github.com/nunit/nunit/pull/2104 to my attention, I had already fixed this one: https://github.com/nunit/nunit/compare/jnm2/values_type_coercion

I'm just going to leave this here until I have time to dig into https://github.com/nunit/nunit/pull/2104. https://github.com/nunit/nunit/pull/2104 looks huge and I don't want to jeopardize it, though perhaps it could be done in granular chunks.

Please comment.

**Comment by ChrisMaddock** [comment]
I agree with everything you've said. Let's discuss #2104 over on #2104. 😄

**Comment by stevenaw** [comment]
I think this was incidentally fixed when resolving https://github.com/nunit/nunit/issues/3053

**Comment by stevenaw** [comment]
I'm closing this issue. I believe it was fixed via #3053 and covered via a test: https://github.com/nunit/nunit/blob/ad49f27294bd0f2677d8699756c6ccb10df600f8/src/NUnitFramework/tests/Attributes/ValuesAttributeTests.cs#L56
